File: backend/dispute/src/main/java/com/commonsengine/dispute/infrastructure/persistence/DisputeEntity.java
package com.commonsengine.dispute.infrastructure.persistence;

import com.commonsengine.dispute.domain.Model.AiScreeningResult;
import com.commonsengine.dispute.domain.Model.ArbitrationVerdict;
import com.commonsengine.dispute.domain.Model.Dispute;
import com.commonsengine.dispute.domain.Model.DisputeId;
import com.commonsengine.dispute.domain.Model.DisputePriority;
import com.commonsengine.dispute.domain.Model.DisputeStatus;
import com.commonsengine.dispute.domain.Model.DisputeType;
import com.commonsengine.dispute.domain.Model.VerdictType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

/**
 * Dispute JPA 实体——与 disputes 表对应。
 *
 * @Entity → 普通 Java 类，无参构造 + getter/setter。
 * 嵌套值对象（AiScreeningResult / ArbitrationVerdict）以 JSON 字符串列存储。
 */
@Entity
@Table(name = "disputes")
public class DisputeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", columnDefinition = "uuid")
    private String id;

    @Column(name = "transaction_id", nullable = false)
    private String transactionId;

    @Column(name = "consumer_id", nullable = false)
    private String consumerId;

    @Column(name = "worker_id", nullable = false)
    private String workerId;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private DisputeType type;

    @Enumerated(EnumType.STRING)
    @Column(name = "priority", nullable = false)
    private DisputePriority priority;

    @Column(name = "description", nullable = false, columnDefinition = "text")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private DisputeStatus status;

    @Column(name = "ai_screening", columnDefinition = "jsonb")
    private String aiScreeningJson;

    @Column(name = "verdict", columnDefinition = "jsonb")
    private String verdictJson;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "resolved_at")
    private Instant resolvedAt;

    public DisputeEntity() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(String transactionId) {
        this.transactionId = transactionId;
    }

    public String getConsumerId() {
        return consumerId;
    }

    public void setConsumerId(String consumerId) {
        this.consumerId = consumerId;
    }

    public String getWorkerId() {
        return workerId;
    }

    public void setWorkerId(String workerId) {
        this.workerId = workerId;
    }

    public DisputeType getType() {
        return type;
    }

    public void setType(DisputeType type) {
        this.type = type;
    }

    public DisputePriority getPriority() {
        return priority;
    }

    public void setPriority(DisputePriority priority) {
        this.priority = priority;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public DisputeStatus getStatus() {
        return status;
    }

    public void setStatus(DisputeStatus status) {
        this.status = status;
    }

    public String getAiScreeningJson() {
        return aiScreeningJson;
    }

    public void setAiScreeningJson(String aiScreeningJson) {
        this.aiScreeningJson = aiScreeningJson;
    }

    public String getVerdictJson() {
        return verdictJson;
    }

    public void setVerdictJson(String verdictJson) {
        this.verdictJson = verdictJson;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getResolvedAt() {
        return resolvedAt;
    }

    public void setResolvedAt(Instant resolvedAt) {
        this.resolvedAt = resolvedAt;
    }
}

File: backend/dispute/src/main/kotlin/com/commonsengine/dispute/infrastructure/persistence/DisputeEntity.kt
package com.commonsengine.dispute.infrastructure.persistence

import com.commonsengine.dispute.domain.Dispute
import com.commonsengine.dispute.domain.DisputeId
import com.commonsengine.dispute.domain.DisputePriority
import com.commonsengine.dispute.domain.DisputeStatus
import com.commonsengine.dispute.domain.DisputeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

/**
 * 纠纷工单 JPA 实体
 *
 * evidenceUrls: List<String> 以分号分隔存储。
 */
@Entity
@Table(name = "disputes")
class DisputeEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "dispute_id", nullable = false, length = 36)
    val disputeId: String,

    @Column(name = "transaction_id", nullable = false, length = 36)
    val transactionId: String,

    @Column(name = "filed_by", nullable = false, length = 36)
    val filedBy: String,

    @Column(name = "filed_against", nullable = false, length = 36)
    val filedAgainst: String,

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false, length = 30)
    val type: DisputeType,

    @Enumerated(EnumType.STRING)
    @Column(name = "priority", nullable = false, length = 15)
    val priority: DisputePriority,

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    val description: String,

    /** 证据 URL 以分号分隔 */
    @Column(name = "evidence_urls", columnDefinition = "TEXT")
    val evidenceUrls: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    var status: DisputeStatus,

    @Column(name = "created_at", nullable = false)
    val createdAt: Instant,

    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant,

    @Column(name = "resolution", columnDefinition = "TEXT")
    var resolution: String? = null,

    @Column(name = "resolved_at")
    var resolvedAt: Instant? = null,
)

// ── Entity ↔ Domain ───────────────────────────────────

fun DisputeEntity.toDomain(): Dispute = Dispute(
    id = DisputeId(disputeId),
    transactionId = transactionId,
    filedBy = filedBy,
    filedAgainst = filedAgainst,
    type = type,
    priority = priority,
    description = description,
    evidenceUrls = evidenceUrls?.split(";")?.filter { it.isNotBlank() } ?: emptyList(),
    status = status,
    createdAt = createdAt,
    updatedAt = updatedAt,
    resolution = resolution,
    resolvedAt = resolvedAt,
)

fun Dispute.toEntity(): DisputeEntity = DisputeEntity(
    disputeId = id.value,
    transactionId = transactionId,
    filedBy = filedBy,
    filedAgainst = filedAgainst,
    type = type,
    priority = priority,
    description = description,
    evidenceUrls = if (evidenceUrls.isEmpty()) null else evidenceUrls.joinToString(";"),
    status = status,
    createdAt = createdAt,
    updatedAt = updatedAt,
    resolution = resolution,
    resolvedAt = resolvedAt,
)
